python/py-xxhash/Portfile

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-xxhash
+version             2.0.0
+platforms           darwin
+license             BSD
+maintainers         nomaintainer
+
+description         Python binding for xxHash
+long_description    ${description}
+
+homepage            https://github.com/ifduyue/python-xxhash
+
+checksums           rmd160  cefd2ef08659a7b3107b2ec8fb27e13866720c8a \
+                    sha256  58ca818554c1476fa1456f6cd4b87002e2294f09baf0f81e5a2a4968e62c423c \
+                    size    64099
+
+python.versions     37 38 39
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
+    depends_lib-append \
+                    port:xxhashlib
+
+    depends_test-append \
+                    port:py${python.version}-pytest
+
+    build.env-append \
+                    XXHASH_LINK_SO=
```

Comment:
I think this was supposed to be `XXHASH_LINK_SO=1`... was there an error you encountered causing you to set this to no value?

Reply:
Doh! Thanks. That corrects linking to `libxxhash`:
```bash
otool -L /opt/local/var/macports/build/_opt_local_ports_python_py-xxhash/py39-xxhash/work/xxhash-2.0.0/build/lib.macosx-10.15-x86_64-3.9/xxhash/_xxhash.cpython-39-darwin.so 
/opt/local/var/macports/build/_opt_local_ports_python_py-xxhash/py39-xxhash/work/xxhash-2.0.0/build/lib.macosx-10.15-x86_64-3.9/xxhash/_xxhash.cpython-39-darwin.so:
	/opt/local/lib/libxxhash.0.dylib (compatibility version 0.0.0, current version 0.8.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)
```

But the import failure in the test is some other issue.

Reply:
For the tests, you can try linking the built library into the library source dir.

This is one horrendously inelegant way to go about it:

```
     pre-test {
        foreach lib_file [glob -nocomplain ${worksrcpath}/build/lib*/xxhash/_xxhash*.*] {
            # symlink built library into Python source directory
            set lib_fname [file tail ${lib_file}]
            system -W ${worksrcpath}/xxhash "ln -s ${lib_file} ${lib_fname}"
        }
     }
```

Reply:
Perhaps not too advisable as the resultant destroot won't necessarily resemble the Python source directory after the above changes...

Reply:
Upstream includes `__init__.py` in the test directory, which breaks the import, a known issue. The latest push fixes this.